pkg/agent/rest.go

```diff
@@ -34,27 +36,38 @@ func NewNodeZeroRestClient(ctx context.Context, assetDir string) (*NodeZeroRestC
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to create asset store")
 	}
-	nmState := &manifests.NMStateConfig{}
-	if err := assetStore.Fetch(nmState); err != nil {
-		return nil, errors.Wrapf(err, "failed to fetch %s", nmState.Name())
+
+	agentConfig, agentConfigError := assetStore.Load(&agentconfig.AgentConfig{})
+	agentManifests, manifestError := assetStore.Load(&manifests.AgentManifests{})
+
+	if agentConfigError != nil {
+		logrus.Debug(errors.Wrapf(agentConfigError, "failed to load %s", agentConfig.(*agentconfig.AgentConfig).Name()))
+	}
+	if manifestError != nil {
+		logrus.Debug(errors.Wrapf(manifestError, "failed to load %s", agentManifests.(*manifests.AgentManifests).Name()))
 	}
 
-	NodeZeroIP, err := manifests.GetNodeZeroIP(nmState.Config)
+	if agentConfigError != nil && manifestError != nil {
```

Comment:
This should be `||` I think.

Reply:
We only need either the agent config or the NM state config but if we don't have either of them we won't have the IP address

Reply:
Not sure I buy that. We only need one, but we don't know which one until we have them both.
This should never fail so any error here is an error I think.

Reply:
I have tested this with only the ZTP manifests and it works as expected. Fails to load the non-existent Agent Config.

Reply:
Can we add unit tests for this function to cover the error scenarios?